### PR TITLE
DROOLS-6812 : Guided rule expression editor text box splitting the expression in multiple text boxes separated by dot(.)

### PR DESCRIPTION
**Pull request by Rikkola**
[DROOLS-6812](https://issues.redhat.com/browse/DROOLS-6812)
[the war](https://drive.google.com/file/d/1s3exW_vKG8J6A-Fzo7pXeD6Cpnu-ISEU/view?usp=sharing)

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest this</b>

* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`

* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

File: drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/Splitter.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                wordBuilder = new StringBuilder();
+                continue;
+            }
+
+            wordBuilder.append(c);
+        }
+
+        result.add(wordBuilder.toString());
+
+        return (String[]) result.toArray(new String[result.size()]);
```
Comment (jomarko) [inline drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/Splitter.java:51]: Interested what sonar says about this part, would you mind converting to regular PR?

Comment (Rikkola) [reply drools-workbench-models/drools-workbench-models-commons/src/main/java/org/drools/workbench/models/commons/backend/rule/Splitter.java:51]: Done

**Comment by jomarko** [comment]
ok but no sonar, maybe rebase and push?

**Comment by Rikkola** [comment]
@jomarko well this is strange. Can't fire it up.

**Comment by Rikkola** [comment]
Jenkins run cdb

**Review by jomarko — APPROVED** [review]
Thank you toni

**Comment by yesamer** [comment]
jenkins retest this please

**Comment by yesamer** [comment]
Jenkins run cdb

**Review by yesamer — APPROVED** [review]

**Comment by Rikkola** [comment]
Jenkins run cdb

**Comment by jomarko** [comment]
Jenkins run cdb

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Can I get a merge please.

**Review by mariofusco — APPROVED** [review]

**Pull request merged**